node/pkg/logscribe/app.go

```diff
@@ -87,3 +148,125 @@ func bulkCopyLogs(ctx context.Context, logsChannel <-chan *[]api.LogInsertModel)
 		}
 	}
 }
+
+func ProcessLogs(ctx context.Context) map[string][]LogInsertModelWithIDWithCount {
+	processedLogs := make(map[string][]LogInsertModelWithIDWithCount)
+	logMap := make(map[string]map[string][]LogInsertModelWithID) // {"service": {hashedLog: []logs}}
+	logs, err := fetchDeleteLogs(ctx)
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to fetch logs")
+		return nil
+	}
+	if len(logs) == 0 {
+		log.Debug().Msgf("No logs to process")
+		return nil
+	}
+
+	for _, log := range logs {
+		hash := hashLog(log)
+		if logMap[log.Service] == nil {
+			logMap[log.Service] = make(map[string][]LogInsertModelWithID)
+		}
+		if logMap[log.Service][hash] == nil {
+			logMap[log.Service][hash] = make([]LogInsertModelWithID, 0)
+		}
+		// uniquely identify logs by hash and store them in a slice to count occurrences later
+		logMap[log.Service][hash] = append(logMap[log.Service][hash], log)
```

Comment:
Why is it storing logs as a list? if it's just for counting I think this logic is overkill. Anyways, this array will have all same entries inside.